server/mysterium_api.go

```diff
@@ -48,10 +48,10 @@ func (mApi *mysteriumApi) RegisterIdentity(identity identity.Identity, signer id
 	return err
 }
 
-func (mApi *mysteriumApi) NodeRegister(proposal dto_discovery.ServiceProposal) error {
-	req, err := newPostRequest("node_register", dto.NodeRegisterRequest{
+func (mApi *mysteriumApi) RegisterProposal(proposal dto_discovery.ServiceProposal, signer identity.Signer) error {
+	req, err := newSignedPostRequest("node_register", dto.NodeRegisterRequest{
```

Comment:
This renaming makes sense! Could you rename the url of discovery-api as well? Calling same endpoint 2 different names looks very confusing.